service/kernel/oneapi/lapack_gpu.cpp

```diff
@@ -79,12 +80,64 @@ services::Status ReferencePotrs<algorithmFPType>::operator()(const math::UpLo up
     return status;
 }
 
+template <typename algorithmFPType>
+services::Status ReferenceXsyevd<algorithmFPType>::operator()(const math::Job jobz, const math::UpLo uplo, const int64_t n,
+                                                              services::Buffer<algorithmFPType> & a, const int64_t lda,
+                                                              services::Buffer<algorithmFPType> & w, services::Buffer<algorithmFPType> & work,
+                                                              const int64_t lwork, services::Buffer<int64_t> & iwork, const int64_t liwork)
+{
+    services::Status status;
+
+    char up = uplo == math::UpLo::Upper ? 'U' : 'L';
+    char job;
+    switch (jobz)
+    {
+    case math::Job::novec: job = 'N'; break;
+    case math::Job::vec: job = 'V'; break;
+    case math::Job::updatevec: job = 'U'; break;
+    case math::Job::allvec: job = 'A'; break;
+    case math::Job::somevec: job = 'S'; break;
+    case math::Job::overwritevec: job = 'O'; break;
+
+    default: status.add(services::UnknownError); break;
+    }
+    DAAL_CHECK_STATUS_VAR(status);
+
+    DAAL_INT nInt      = static_cast<DAAL_INT>(n);
+    DAAL_INT ldaInt    = static_cast<DAAL_INT>(lda);
+    DAAL_INT lworkInt  = static_cast<DAAL_INT>(lwork);
+    DAAL_INT liworkInt = static_cast<DAAL_INT>(liwork);
+
+    services::SharedPtr<algorithmFPType> aPtr    = a.toHost(data_management::ReadWriteMode::readWrite);
+    services::SharedPtr<algorithmFPType> wPtr    = w.toHost(data_management::ReadWriteMode::readWrite);
+    services::SharedPtr<algorithmFPType> workPtr = work.toHost(data_management::ReadWriteMode::readWrite);
+    services::SharedPtr<int64_t> iworkSharedPtr        = iwork.toHost(data_management::ReadWriteMode::readWrite);
+
+    DAAL_INT info;
+
+    services::internal::TArray<DAAL_INT, CpuType::sse2> iWorkTmp(iwork.size());
+
+    LapackAutoDispatch<algorithmFPType>::xsyevd(&job, &up, &nInt, aPtr.get(), &ldaInt, wPtr.get(), workPtr.get(), &lworkInt, iWorkTmp.get(),
+                                                &liworkInt, &info);
+    DAAL_CHECK(info == 0, services::ErrorID::ErrorNormEqSystemSolutionFailed);
+
+    int64_t* iworkPtr = iworkSharedPtr.get();
+    for (size_t i = 0; i < iwork.size(); ++i) {
+        iworkPtr[i] = static_cast<int64_t>(iWorkTmp[i]);
+    }
```

Comment:
can we avoid such conversions? Does Lapack have different APIs for GPU and CPU?

Reply:
In the end CPU requires `DAAL_INT *`, but GPU needs `int64_t`

Reply:
Which type is wider?